dotnet/sdk
# dotnet tool install skip unlisted version

Fixes #24037 
Use the `NugetToolSearchApiRequest` to get the latest listed package version in `dotnet tool install <PACKAGE_ID>` when `--version` is not supplied.

### Manual Testing
Created a test tool `jackywang118.botsay` with two versions where the most recent version `2.0.0` is unlisted. 
Run `dotnet tool install jackywang118.botsay`, version `1.0.0` should be installed
A request to [https://azuresearch-usnc.nuget.org/query?q=jackywang118.botsay&packageType=dotnettool&semVerLevel=2.0.0](https://azuresearch-usnc.nuget.org/query?q=jackywang118.botsay&packageType=dotnettool&semVerLevel=2.0.0) is made to get the most recent listed package version
